xlr/sdk/src/validator.ts

```diff
@@ -16,16 +16,29 @@ import {
   resolveReferenceNode,
   computeEffectiveObject,
 } from "@player-tools/xlr-utils";
-import type { ValidationError } from "./types";
+import { ValidationSeverity } from "./types";
+import type { ValidationMessage } from "./types";
+
+export interface XLRValidatorConfig {
+  /** URL mapping for supplemental documentation */
+  urlMapping?: Record<string, string>;
+}
+
+const MAX_VALID_SHOWN = 20;
 
 /**
  * Validator for XLRs on JSON Nodes
  */
 export class XLRValidator {
+  private config: XLRValidatorConfig;
   private resolveType: (id: string) => NamedType<NodeType> | undefined;
   private regexCache: Map<string, RegExp>;
 
-  constructor(resolveType: (id: string) => NamedType<NodeType> | undefined) {
+  constructor(
+    resolveType: (id: string) => NamedType<NodeType> | undefined,
+    config?: XLRValidatorConfig
```

Comment:
Right now since the Validator is private on the SDK there wouldn't be a way to load a config in to the validator. We would probably need to extend the SDK to take config in for the validator to pass down as well as expose it to CLI plugins. _Technically_ I don't think its a blocker but something we should make a follow up for.

Reply:
Ugh okay, didn't realize that was a limitation but yea sounds good since ideally we'd want to pass in those URL's into cg-wp.

Reply:
https://github.com/player-ui/tools/issues/194